File: identity-service/src/main/java/com/example/identity_service/dto/response/ProductResponse.java
package com.example.identity_service.dto.response;

import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProductResponse {
    Integer productId;

    String categoryName;

    String productName;

    String description;

    BigDecimal price;

    Integer stockQuantity;

    String imageUrl;

}

File: computer-selling-v1/src/app/model/response/product.response.ts
import { Category } from "./category.response"

export interface ProductItemResponse {
    productId: number
    productName: string
    category: Category
    description: string
    price: number
    stockQuantity: number
    imageUrl: string
    createdAt: string
    promotion: any
    categoryName: string
    categoryId: string
}